File: packages/modules/sms/ios/Sms.m
// Copyright 2026 The Lynxpo Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

#import "Sms.h"
#import <Foundation/Foundation.h>

@implementation Sms

+ (NSString *)name {
 return @"Sms";
}

+ (NSDictionary<NSString *, NSString *> *)methodLookup {
 return @{
  @"isAvailable" : NSStringFromSelector(@selector(isAvailable)),
  @"sendSMS" : NSStringFromSelector(@selector(sendSMSWithAddresses:(NSArray *)addresses message:(NSString *)message)),
 };
}

- (BOOL)isAvailable {
 return YES;
}

- (void)sendSMSWithAddresses:(NSArray *)addresses message:(NSString *)message {
}

@end

File: packages/modules/sms/android/src/main/java/com/lynxpo/sms/Sms.java
// Copyright 2026 The Lynxpo Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.
package com.lynx.explorer.modules;

import android.content.Context;
import com.lynx.jsbridge.LynxMethod;
import com.lynx.jsbridge.LynxModule;
import com.lynx.jsbridge.LynxNativeModule;
import com.lynxpo.sms.generated.SmsSpec;
import com.lynx.react.bridge.ReadableArray;
import com.lynx.react.bridge.WritableMap;
import com.lynx.react.bridge.JavaOnlyMap;
import com.lynx.react.bridge.WritableArray;
import com.lynx.react.bridge.JavaOnlyArray;

/**
 * Android counterpart of the iOS {@code Sms}. Exposes functionality to JS via
 * {@code NativeModules.Sms}, faithfully porting Expo's native method surface.
 */
@LynxNativeModule(name = "Sms")
public class Sms extends SmsSpec {

  public Sms(Context context) {
    super(context);
  }

  @LynxMethod
  public boolean isAvailable() {
    return true;
  }

  @LynxMethod
  public void sendSMS(ReadableArray addresses, String message) {
  }

}
